kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js

```diff
@@ -323,52 +387,61 @@ define([
                 node: createTableRow(tableBody, jobState.job_id),
                 jobState: jobState,
                 // this will be replaced once the job-info call runs
-                name: jobState.description || 'Child Job ' + (jobIndex + 1),
+                name: jobState.description || jobState.job_id || 'Child Job ' + (jobIndex + 1),
                 clickAction: doSingleJobAction,
             });
         }
 
         /**
          *
-         * @param {object} args  -- with keys
+         * @param {object} args  -- with key
          *      node:     DOM node to attach to
-         *      jobState: job state object
-         *      includeParentJob: boolean; whether or not to add a parent job row
          *
          * @returns {Promise} started JobStateList widget
          */
         function start(args) {
-            const requiredArgs = ['node', 'jobState'];
+            const requiredArgs = ['node'];
             if (!requiredArgs.every((arg) => arg in args && args[arg])) {
                 throw new Error('start argument must have these keys: ' + requiredArgs.join(', '));
             }
+
+            const indexedJobs = model.getItem('exec.jobs.byId');
+            if (!indexedJobs || !Object.keys(indexedJobs).length) {
+                throw new Error('Must provide at least one job to show the job state list');
+            }
+
             const events = Events.make({ node: args.node });
-            $container = $(args.node);
+            container = args.node;
             const actionDropdown = createActionsDropdown(events);
 
-            $container.append($(actionDropdown + '\n' + createTable()));
-            [tableBody] = $container.find('tbody');
-            const jobState = Jobs.updateJobModel(args.jobState);
+            container.innerHTML = [
+                div(
+                    {
+                        class: `${cssBaseClass}__dropdown_container`,
+                    },
+                    actionDropdown
+                ),
+                createTable(),
+            ].join('\n');
+
+            tableBody = container.querySelector('tbody');
+            const jobs = Object.values(indexedJobs);
 
             return Promise.all(
-                jobState.child_jobs.map((childJob, index) => {
-                    createJobStateListRowWidget(childJob, index);
+                jobs.map((jobState, index) => {
+                    createJobStateListRowWidget(jobState, index);
                 })
             ).then(() => {
-                renderTable($container, jobState.child_jobs.length);
+                renderTable($(container), jobs.length);
                 events.attachEvents();
 
-                // TODO: depending on which strategy we use for updating the job status
-                // table, we can remove either the parent listener or that for the
-                // child jobs
```

Comment:
parent job is not very useful, especially once jobs are completed, so the widget tracks the status of the child jobs